### PR TITLE
[Improvement]: Using unified catalog for server external catalog.

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?

This is a subtask of https://github.com/NetEase/amoro/issues/1061. 

In this PR, the AMS will use UnifiedCatalog as ExternalCatalog implement.

## Brief change log
 
- Implement MixedIceberg FormatCatalog
- Using UnifiedCatalog for ExternalCatalog implement in AMS

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? ( no)

### CODE REVIEW

File: core/src/main/java/com/netease/arctic/CommonUnifiedCatalog.java [hunk 1]
```diff
@@ -42,8 +42,10 @@ public class CommonUnifiedCatalog implements UnifiedCatalog {
...
   public CommonUnifiedCatalog(
-      Supplier<CatalogMeta> catalogMetaSupplier, CatalogMeta meta, Map<String, String> properties) {
-    this.meta = meta;
+      Supplier<CatalogMeta> catalogMetaSupplier, Map<String, String> properties) {
+    CatalogMeta catalogMeta = catalogMetaSupplier.get();
+    CatalogUtil.mergeCatalogProperties(catalogMeta, properties);
+    this.meta = catalogMeta;
     this.properties.putAll(properties);
     this.metaSupplier = catalogMetaSupplier;
     initializeFormatCatalogs();
```
Comment (wangtaohz) [inline core/src/main/java/com/netease/arctic/CommonUnifiedCatalog.java:51]: Do you think this `initializeFormatCatalogs` method should be invoked when calling `com.netease.arctic.CommonUnifiedCatalog#refresh`?

**Comment by shidayang** [comment]
BTW, Your need test in local kerberos env

**Comment by shidayang** [comment]
Need add UT for AmoroCatalog of mixed-format

File: core/src/main/java/com/netease/arctic/catalog/ArcticCatalog.java [hunk 2]
```diff
@@ -52,6 +53,8 @@ public interface ArcticCatalog {
    */
   void initialize(AmsClient client, CatalogMeta meta, Map<String, String> properties);
 
+  void initialize(String name, Map<String, String> properties, TableMetaStore metaStore);
```
Comment (wangtaohz) [inline core/src/main/java/com/netease/arctic/catalog/ArcticCatalog.java:56]: This `initialize` method needs some docs.

Additionally, I'm a bit confused about when to use these two initialize methods and whether the properties refer to client properties or catalog properties. It would be better to differentiate them in the parameter names and docs.

File: ams/server/src/main/java/com/netease/arctic/server/catalog/ExternalCatalog.java [hunk 3]
```diff
@@ -1,14 +1,31 @@
...
+import java.util.stream.Collectors;
+
+public class ExternalCatalog extends ServerCatalog {
+
+  UnifiedCatalog unifiedCatalog;
+  TableMetaStore tableMetaStore;
 
   protected ExternalCatalog(CatalogMeta metadata) {
     super(metadata);
+    this.unifiedCatalog = new CommonUnifiedCatalog(this::getMetadata, Maps.newHashMap());
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply ams/server/src/main/java/com/netease/arctic/server/catalog/ExternalCatalog.java]: fixed

File: core/src/main/java/com/netease/arctic/CommonUnifiedCatalog.java (hunk 1 above)
Comment (baiyangtx) [reply core/src/main/java/com/netease/arctic/CommonUnifiedCatalog.java:51]: fixed

File: core/src/main/java/com/netease/arctic/catalog/ArcticCatalog.java (hunk 2 above)
Comment (baiyangtx) [reply core/src/main/java/com/netease/arctic/catalog/ArcticCatalog.java:67]: this method is deprecated.

Comment (baiyangtx) [reply core/src/main/java/com/netease/arctic/catalog/ArcticCatalog.java:52]: I have been removed this method.

File: core/src/main/java/com/netease/arctic/catalog/BasicIcebergCatalog.java [hunk 4]
```diff
@@ -44,9 +40,9 @@ public String name() {
   }
 
   @Override
-  public void initialize(AmsClient client, CatalogMeta meta, Map<String, String> properties) {
-    this.client = client;
-    this.catalogWrapper = new IcebergCatalogWrapper(meta, properties);
+  public void initialize(String name, Map<String, String> properties, TableMetaStore metaStore) {
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply core/src/main/java/com/netease/arctic/catalog/BasicIcebergCatalog.java]: fixed.

File: core/src/main/java/com/netease/arctic/catalog/ArcticCatalog.java [hunk 5]
```diff
@@ -46,11 +45,11 @@ public interface ArcticCatalog {
...
    * must be no-args construct. Catalogs will call this method after implement object created.
    *
-   * @param client client of arctic metastore
-   * @param meta catalog init struct
+   * @param name name of catalog
    * @param properties client side catalog properties
+   * @param metaStore auth context.
    */
-  void initialize(AmsClient client, CatalogMeta meta, Map<String, String> properties);
+  void initialize(String name, Map<String, String> properties, TableMetaStore metaStore);
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply core/src/main/java/com/netease/arctic/catalog/ArcticCatalog.java:52]: I will do this in anthor PR

File: ams/server/src/main/java/com/netease/arctic/server/catalog/CatalogBuilder.java [hunk 6]
```diff
@@ -11,45 +11,60 @@
...
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
 
+import java.util.Map;
 import java.util.Set;
 
 public class CatalogBuilder {
 
-  // TODO: use internal or external concepts
+  /** matrix of catalog type and supported table formats */
+  private static final Map<String, Set<TableFormat>> formatSupportedMatrix =
```
[the remark this answers is not in the document]
Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/catalog/CatalogBuilder.java:24]: I also think these two parts are somewhat repetitive. 

BTW, for the variable name of "static final", it is better to use capital letters with underscores.

File: ams/server/src/test/java/com/netease/arctic/server/AmsEnvironment.java [hunk 7]
```diff
@@ -203,9 +203,9 @@ private void createInternalIceberg() {
             TableFormat.ICEBERG);
 
     tableService.createCatalog(catalogMeta);
-    catalogs.put(
-        INTERNAL_ICEBERG_CATALOG,
-        CatalogLoader.load(getTableServiceUrl() + "/" + INTERNAL_ICEBERG_CATALOG));
+    // catalogs.put(
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply ams/server/src/test/java/com/netease/arctic/server/AmsEnvironment.java]: removed.

File: core/src/test/java/com/netease/arctic/catalog/TableTestBase.java [hunk 8]
```diff
@@ -76,14 +78,30 @@ private void createIcebergFormatTable() {
...
+        (ArcticTable)
+            getUnifiedCatalog()
+                .loadTable(TableTestHelper.TEST_DB_NAME, TableTestHelper.TEST_TABLE_NAME)
+                .originalTable();
   }
 
   @After
   public void dropTable() {
-    getCatalog().dropTable(tableTestHelper.id(), true);
+    switch (getTestFormat()) {
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply core/src/test/java/com/netease/arctic/catalog/TableTestBase.java]: fixed.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Review by wangtaohz — APPROVED** [review]
